File: client/src/dtos/RoomStateDto.ts
import QueuedMediaDto from "./QueuedMediaDto"

type RoomStateDto = {
    roomId: string
    displayName: string
    owner: string
    visibility: "public" | "unlisted"
    connectedUsers: string[]
    queue: QueuedMediaDto[]
    currentlyPlayingMedia: QueuedMediaDto | undefined
    currentlyPlayingMediaStartedAt: string | undefined
}

export default RoomStateDto

File: server/src/main/kotlin/com/example/pojos/RoomStateDto.kt
package com.example.pojos

import kotlinx.serialization.Serializable

@Serializable
data class RoomStateDto(
    val roomId: String,
    val displayName: String,
    val connectedUsers: Set<String>,
    val queue: Set<QueuedMediaDto>,
    val currentlyPlayingMedia: QueuedMediaDto?,
    val currentlyPlayingMediaStartedAt: String?,
    val owner: String,
    val visibility: String
)